src/assets/wise5/classroomMonitor/dataExport/strategies/AbstractComponentDataExportStrategy.ts

```diff
@@ -0,0 +1,398 @@
+    this.setStudentIDsAndNames(
+      row,
+      columnNameToNumber,
+      userId1,
+      studentName1,
+      userId2,
+      studentName2,
+      userId3,
+      studentName3
+    );
+    row[columnNameToNumber.get('Class Period')] = periodName;
+    row[columnNameToNumber.get('Project ID')] = this.configService.getProjectId();
+    row[columnNameToNumber.get('Project Name')] = this.projectService.getProjectTitle();
+    row[columnNameToNumber.get('Run ID')] = this.configService.getRunId();
+    row[columnNameToNumber.get('Student Work ID')] = componentState.id;
+    if (componentState.serverSaveTime != null) {
+      const formattedDateTime = this.utilService.convertMillisecondsToFormattedDateTime(
+        componentState.serverSaveTime
+      );
+      row[columnNameToNumber.get('Server Timestamp')] = formattedDateTime;
+    }
+    if (componentState.clientSaveTime != null) {
+      const clientSaveTime = new Date(componentState.clientSaveTime);
+      if (clientSaveTime != null) {
+        const clientSaveTimeString =
+          clientSaveTime.toDateString() + ' ' + clientSaveTime.toLocaleTimeString();
+        row[columnNameToNumber.get('Client Timestamp')] = clientSaveTimeString;
+      }
+    }
+    row[columnNameToNumber.get('Node ID')] = componentState.nodeId;
+    row[columnNameToNumber.get('Component ID')] = componentState.componentId;
+    row[columnNameToNumber.get('Step Title')] = this.projectService.getNodePositionAndTitle(
+      componentState.nodeId
+    );
+    const componentPartNumber =
+      this.projectService.getComponentPosition(componentState.nodeId, componentState.componentId) +
+      1;
+    row[columnNameToNumber.get('Component Part Number')] = componentPartNumber;
+    row[columnNameToNumber.get('Component Type')] = this.component.type;
+    if (this.component.prompt != null) {
+      let prompt = this.utilService.removeHTMLTags(this.component.prompt);
+      prompt = prompt.replace(/"/g, '""');
+      row[columnNameToNumber.get('Component Prompt')] = prompt;
+    }
+    const studentData = componentState.studentData;
+    if (studentData != null) {
+      row[columnNameToNumber.get('Student Data')] = studentData;
+      const isCorrect = studentData.isCorrect;
+      if (isCorrect != null) {
+        if (isCorrect) {
+          row[columnNameToNumber.get('Is Correct')] = 1;
+        } else {
+          row[columnNameToNumber.get('Is Correct')] = 0;
+        }
+      }
+    }
+    const revisionCounter = this.getRevisionCounter(
+      componentRevisionCounter,
+      componentState.nodeId,
+      componentState.componentId
+    );
+    if (componentState.revisionCounter == null) {
+      /*
+       * use the revision counter obtained from the componentRevisionCounter
+       * mapping. this case will happen when we are exporting all student
+       * work.
+       */
+      row[columnNameToNumber.get('Component Revision Counter')] = revisionCounter;
```

Comment:
```revisionCounter``` is only used in the ```if (componentState.revisionCounter == null)``` block, so move it there?
```suggestion
    if (componentState.revisionCounter == null) {
      /*
       * use the revision counter obtained from the componentRevisionCounter
       * mapping. this case will happen when we are exporting all student
       * work.
       */
      row[columnNameToNumber.get('Component Revision Counter')] =  this.getRevisionCounter(
      componentRevisionCounter,
      componentState.nodeId,
      componentState.componentId
    );
```